pkg/core/stateroot/prometheus.go

```diff
@@ -0,0 +1,20 @@
+package stateroot
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// stateHeight prometheus metric.
+var stateHeight = prometheus.NewGauge(
+	prometheus.GaugeOpts{
+		Help:      "Current verified state height",
+		Name:      "current_state_height",
+		Namespace: "neogo",
+	},
+)
+
+func init() {
+	prometheus.MustRegister(stateHeight)
+}
+
+func updateStateHeightMetric(sHeight uint32) {
```

Comment:
Is it called from anywhere?

Reply:
We need to always update it with in-header option and update for validated roots otherwise.

Reply:
Fixed.